### PR TITLE
Refactor(eos_designs)!: Change L2 uplink description

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Change L2 uplink description
## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Adding new default descriptions for L2 uplinks (port-channels and members)
- Add templating support for all
- Add keys with templates:
  - `underlay_l2_ethernet_description`
  - `underlay_l2_port_channel_description`
- Updated porting guide

## How to test

- Update molecule scenarios and examples
- Verified no changes for scenarios with custom description logic in unit tests

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/api/interface_descriptions/__init__.py [hunk 1]
```diff
@@ -498,7 +497,7 @@ class InterfaceDescriptionData:
     """Interface of peer"""
     peer_channel_group_id: int | None
     """Port channel ID of peer"""
-    peer_mlag_group: str | None
+    peer_node_group: str | None
     """MLAG group of peer"""
```
Comment (gmuloc) [inline python-avd/pyavd/api/interface_descriptions/__init__.py]:
```suggestion
    peer_node_group: str | None
    """Node group of peer"""
```

File: python-avd/pyavd/api/interface_descriptions/__init__.py [hunk 2]
```diff
@@ -101,7 +99,7 @@ def underlay_port_channel_interface(self, data: InterfaceDescriptionData) -> str
             - peer
             - peer_interface
             - peer_channel_group_id
-            - peer_mlag_group
+            - peer_group
```
Comment (gmuloc) [inline python-avd/pyavd/api/interface_descriptions/__init__.py]:
```suggestion
            - peer_node_group
```

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/underlay_l2_port_channel_description.schema.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      The description or description template to be used on L2 port-channel interfaces.
+      The interfaces using this are port-channel uplinks.
+      This can be a template using the format string syntax.
+      The available template fields are:
+        - `peer`: The name of the peer.
+        - `interface`: The local interface name.
+        - `peer_interface`: The interface on the peer.
+        - `port_channel_id`: The local port-channel ID.
+        - `peer_port_channel_id`: The ID of the port-channel on the peer.
+        - `peer_node_group`: The node group of the peer if the peer is an MLAG member or running EPVN A/A.
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/underlay_l2_port_channel_description.schema.yml]:
```suggestion
        - `peer_node_group`: The node group of the peer if the peer is an MLAG member or running EVPN A/A.
```

**Review by gmuloc — APPROVED** [review]
LGTM after CI complete (pre-commit needs to re run as I just fixed a typo in the schema)

**Review by carlbuchmann — APPROVED** [review]
LGTM - tested reverting back to 4.x syntax in local inventory.

**Pull request merged**